Firestore/Source/Core/FSTSyncEngine.mm

```diff
@@ -344,15 +342,16 @@ - (void)applyChangedOnlineState:(FSTOnlineState)onlineState {
   [self.delegate handleViewSnapshots:newViewSnapshots];
 }
 
-- (void)rejectListenWithTargetID:(FSTBoxedTargetID *)targetID error:(NSError *)error {
+- (void)rejectListenWithTargetID:(const TargetId)targetID error:(NSError *)error {
   [self assertDelegateExistsForSelector:_cmd];
 
-  FSTDocumentKey *limboKey = self.limboKeysByTarget[targetID];
-  if (limboKey) {
+  const auto iter = _limboKeysByTarget.find(targetID);
+  if (iter != _limboKeysByTarget.end()) {
+    const DocumentKey limboKey = iter->second;
```

Comment:
Alternative to lines 348-350:

```
if (_limboKeysByTarget.find(targetID) != _limboKeysByTarget.end()) {
  const DocumentKey limboKey = _limboKeysByTarget[targetID];
  ...
```

Optional. (I think it reads a bit better this way, but YMMV. Feel free to leave as is.)